File: src/main/java/com/dh/ondot/member/core/exception/NotFoundMemberException.java
package com.dh.ondot.member.core.exception;

import com.dh.ondot.core.exception.NotFoundException;

import static com.dh.ondot.core.exception.ErrorCode.NOT_FOUND_MEMBER;

public class NotFoundMemberException extends NotFoundException {
  public NotFoundMemberException(Long memberId) {
    super(NOT_FOUND_MEMBER.getMessage().formatted(memberId));
  }

  @Override
  public String getErrorCode() {
    return NOT_FOUND_MEMBER.name();
  }
}

File: src/main/kotlin/com/dh/ondot/member/core/exception/NotFoundMemberException.kt
package com.dh.ondot.member.core.exception

import com.dh.ondot.core.exception.ErrorCode.NOT_FOUND_MEMBER
import com.dh.ondot.core.exception.NotFoundException

class NotFoundMemberException(memberId: Long) :
    NotFoundException(NOT_FOUND_MEMBER.message.format(memberId)) {
    override val errorCode: String get() = NOT_FOUND_MEMBER.name
}
